### PR TITLE
Migrate unique token store to use a VirtualMap instead of MerkleMap

### PR DESCRIPTION
**Pull request by rjuang**
**Description**:

This PR migrates the unique token store from a MerkleMap to a VirtualMap, replacing all usages of the MerkleMap with the corresponding VirtualMap method calls. It also adds the required migration needed to update a node's data holdings from MerkleMap to VirtualMap. Note: this PR includes https://github.com/hashgraph/hedera-services/pull/2980 and is recommended that that PR is submitted first before this one.

**Related issue(s)**:

Fixes: https://github.com/hashgraph/hedera-services/issues/2820

**Notes for reviewer**:
- This PR is branched from PR https://github.com/hashgraph/hedera-services/pull/2980
  - The new UniqueToken*.java files and pom.xml file changes are all in that PR. 
- Majority of the migration logic is in:
  - ReleaseTwentyFive.java
  - ServicesState.java
- All other files are related to switching from MerkleMap to VirtualMap for integration.
- TODO performance numbers
- 
**Checklist**
- [ ] Tested (unit, integration, etc.)
- [ ] Check performance results

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/ServicesState.java [hunk 1]
```diff
@@ -164,14 +167,22 @@ public void addDeserializedChildren(List<MerkleNode> children, int version) {
...
 		int deserializedVersionFromState = getDeserializedVersion();
+		boolean migrationProcessed = false;
 		if (deserializedVersionFromState < RELEASE_0240_VERSION) {
 			stakeFundingMigrator.accept(this);
 			if (deserializedVersionFromState < RELEASE_0220_VERSION) {
 				// Remove after confirming no regression tests use 0.21.x states
-				blobMigrator.migrateFromBinaryObjectStore(this, deserializedVersionFromState);
-				init(getPlatformForDeferredInit(), getAddressBookForDeferredInit(), getDualStateForDeferredInit());
+				blobMigrator.migrate(this, deserializedVersionFromState);
+				migrationProcessed = true;
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/ServicesState.java:176]: The curious-looking "deferred `init()`" pattern is unnecessary; the `0.22` migration touched data stored in PostgreSQL and that's why it was required then.

File: hedera-node/src/main/java/com/hedera/services/utils/EntityNum.java [hunk 2]
```diff
@@ -47,22 +45,21 @@
 public class EntityNum implements Comparable<EntityNum> {
 	public static final EntityNum MISSING_NUM = new EntityNum(0);
 
-	private final int value;
+	private final long value;
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/utils/EntityNum.java:48]: When we replace this with a `long`, imo we should simultaneously migrate the `number` fields of all the types that appear as values in a `MerkleMap` with `EntityNum` key. (For example, it's no longer _technically_ safe for `MerkleAccountState#number` to be an `int`---although in practice we could get away with this for a very long time.)

Comment (rjuang) [reply hedera-node/src/main/java/com/hedera/services/utils/EntityNum.java:48]: I'm wondering if the int vs num issue might be causing an issue. Maybe I should focus on migrating this to long later on, but not here.. Let me revert the changes to long.

File: hedera-node/src/main/java/com/hedera/services/ServicesState.java (hunk 1 above)
Comment (rjuang) [reply hedera-node/src/main/java/com/hedera/services/ServicesState.java:176]: If I don't defer the init() pattern, then I get a "ERROR: Signed state loaded from disk has an invalid hash!"

File: hedera-node/src/main/java/com/hedera/services/utils/EntityNum.java (hunk 2 above)
Comment (rjuang) [reply hedera-node/src/main/java/com/hedera/services/utils/EntityNum.java]: Reverted EntityNum changes and will postpone this to subsequent release. Hopefully, this fixes the issues seen.

File: hedera-node/src/main/java/com/hedera/services/state/migration/ReleaseTwentyFiveMigration.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package com.hedera.services.state.migration;
```
Comment (rjuang) [inline hedera-node/src/main/java/com/hedera/services/state/migration/ReleaseTwentyFiveMigration.java:1]: TODO: Add licensing comment block

File: hedera-node/src/test/java/com/hedera/services/state/migration/ReleaseTwentyFiveMigrationTest.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
+package com.hedera.services.state.migration;
```
Comment (rjuang) [inline hedera-node/src/test/java/com/hedera/services/state/migration/ReleaseTwentyFiveMigrationTest.java:1]: TODO: Add license comment block

File: hedera-node/src/main/java/com/hedera/services/state/virtual/VirtualMapFactory.java [hunk 5]
```diff
@@ -98,4 +100,28 @@ public VirtualMap<ContractKey, ContractValue> newVirtualizedStorage() {
 				.internalHashesRamToDiskThreshold(MAX_IN_MEMORY_INTERNAL_HASHES);
 		return new VirtualMap<>(STORAGE_VM_NAME, dsBuilder);
 	}
+
+	public VirtualMap<UniqueTokenKey, UniqueTokenValue> newVirtualizedUniqueTokenStorage() {
+		var storageKeySerializer = new UniqueTokenKeySerializer();
```
Comment (rjuang) [inline hedera-node/src/main/java/com/hedera/services/state/virtual/VirtualMapFactory.java:105]: rename to uniqueTokenKeySerializer

File: hedera-node/src/main/java/com/hedera/services/state/virtual/VirtualMapFactory.java [hunk 6]
```diff
@@ -98,4 +100,28 @@ public VirtualMap<ContractKey, ContractValue> newVirtualizedStorage() {
...
+		VirtualLeafRecordSerializer<UniqueTokenKey, UniqueTokenValue> storageLeafRecordSerializer =
```
Comment (rjuang) [inline hedera-node/src/main/java/com/hedera/services/state/virtual/VirtualMapFactory.java:106]: rename to uniqueTokenLeafRecordSerializer

**Comment by rjuang** [comment]
Deprecating this with https://github.com/hashgraph/hedera-services/pull/3407

**Pull request closed without merging**